tests/functional/configs/test_grant_configs.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+
+from dbt.tests.util import run_dbt, get_manifest, write_file
+
+dbt_project_yml = """
+models:
+  test:
+    my_model:
+      +grants:
+        my_select: ["reporter", "bi"]
+"""
+
+model_schema_yml = """
+version: 2
+models:
+  - name: my_model
+    config:
+      grants:
+        my_select: ["someone"]
+"""
+
+
+my_model_sql = """
+{{ config(grants={'my_select': ['other_user']}) }}
+select 1 as fun
+"""
+
+
+class TestGrantConfigs:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"my_model.sql": my_model_sql}
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return dbt_project_yml
+
+    def test_model_grant_config(self, project):
+        # This test uses "my_select" instead of "select", so that when
+        # actual granting of permissions happens, it won't break this
+        # test.
+        results = run_dbt(["run"])
+        assert len(results) == 1
+
+        manifest = get_manifest(project.project_root)
+        model_id = "model.test.my_model"
+        assert model_id in manifest.nodes
+
+        model = manifest.nodes[model_id]
+        model_config = model.config
+        assert hasattr(model_config, "grants")
+
+        expected = {"my_select": ["reporter", "bi", "other_user"]}
```

Comment:
There needs to be some mechanism to remove grants that have been applied at a broader level of the configuration hierarchy.

So: I think we need `other_user` to clobber the existing values of the `my_select` key (`["reporter", "bi"]`), rather than append to the list of recipients for the `my_select` privilege.

The ideal case is one in which users have some control over this merging behavior. Is now a good time to revisit the discussion in https://github.com/dbt-labs/dbt-core/issues/4108? (Even if we don't / can't actually enable that now)